src/foam/core/FObject.java

```diff
@@ -295,12 +295,24 @@ default FObject copyFrom(FObject obj) {
     List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
     for ( PropertyInfo p : props ) {
       try {
-        if ( p.isSet(obj) ) p.set(this, p.get(obj));
+        if ( p.isSet(obj) ) {
+          var value = p.get(obj);
+          if ( value instanceof FObject ) {
+            value = ((FObject) p.get(this)).copyFrom((FObject) value);
+          }
+          p.set(this, value);
+        }
```

Comment:
JS supports nested copyFrom (https://github.com/kgrgreer/foam3/blob/884f5234fd39842a59ac3abc1cc266698d6626a0/src/foam/core/FObject.js#L987-L990 and https://github.com/kgrgreer/foam3/blob/5afd2ca120cdef410bd73b68d33f7fb985c63c0f/src/foam/core/types.js#L860-L867).

Might need to have the equivalent of copyValueFrom() on PropertyInfo for Java. @kgrgreer @jlhughes @nanoscott any though?